modules/programs/alot.nix

```diff
@@ -0,0 +1,145 @@
+# alot config loader is sensitive to leading space !
+{ config, lib, pkgs, ... } @ top:
+
+with lib;
+with import ../lib/dag.nix { inherit lib; };
+
+let
+
+  cfg = config.programs.alot;
+
+  notmuchConfig = account: "$XDG_CONFIG_HOME/notmuch/notmuchrc";
+
+  alot_accounts = config.accounts.email.accounts;
```

Comment:
Maybe need to filter out those accounts for which `alot.enable == true`? Something like

```nix
filter (a: a.alot.enable) (attrValues config.accounts.email.accounts)
```

Reply:
I'm guessing alot requires notmuch to be enabled for the account? If so, then it's probably best to add an assertion along the lines of

```nix
assertions = [
  (
    let
      badAccounts = filter (a: !a.notmuch.enable) alotAccounts;
    in
      {
        assertion = badAccounts == [];
        message = "alot: Notmuch not enabled for accounts: "
          + concatMapStringsSep ", " (a: a.name) badAccounts;
      }
  )

  {
    assertion = config.notmuch.enable;
    message = "Notmuch not enabled";
  }
];
```

Note, not tested and the message phrasing could be improved.

Reply:
I would rather have alot enable notmuch automatically ?!  (yes notmuch is mandatory)